a8c-files.php

```diff
@@ -381,60 +367,23 @@ function upload_file( $details, $upload_type ) {
 			$post_url = $this->get_files_service_hostname() . $file_path;
 		}
 
-		$headers = array(
-					'X-Client-Site-ID: ' . constant( 'FILES_CLIENT_SITE_ID' ),
-					'X-Access-Token: ' . constant( 'FILES_ACCESS_TOKEN' ),
-					'Content-Type: ' . $details['type'],
-					'Content-Length: ' . filesize( $details['file'] ),
-					'Connection: Keep-Alive',
-				);
-
-		$stream = fopen( $details['file'], 'r' );
-		$ch = curl_init( $post_url );
-
-		curl_setopt( $ch, CURLOPT_PUT, true );
-		curl_setopt( $ch, CURLOPT_INFILE, $stream );
-		curl_setopt( $ch, CURLOPT_INFILESIZE, filesize( $details['file'] ) );
-		curl_setopt( $ch, CURLOPT_RETURNTRANSFER, true );
-		curl_setopt( $ch, CURLOPT_HEADER, false );
-		curl_setopt( $ch, CURLOPT_HTTPHEADER, $headers );
-		curl_setopt( $ch, CURLOPT_TIMEOUT, 10 + (int)( filesize( $details['file'] ) / 512000 ) ); // 10 plus 1 second per 500k
-
-		curl_setopt( $ch, CURLOPT_READFUNCTION,
-					function( $ch, $fd, $length ) use( $stream ) {
-						$data = fread( $stream, $length );
-						if ( null == $data )
-							return 0;
-						else
-							return $data;
-					});
-
-		$ret_data = curl_exec( $ch );
-		$http_code = curl_getinfo( $ch, CURLINFO_HTTP_CODE );
-		curl_close( $ch );
-		fclose( $stream );
+		$upload_result = $this->api_client->upload_file( $details['file'], $file_path );
+
 		register_shutdown_function( 'unlink', $details['file'] );
 
-		switch ( $http_code ) {
-			case 200:
-				if ( 0 < strlen( $ret_data ) ) {
-					$obj = json_decode( $ret_data );
-					if ( isset(  $obj->filename ) && basename( $obj->filename ) != basename( $post_url ) ) {
-						$uploads = wp_upload_dir();
-						if ( false === $uploads['error'] ) {
-							@copy( $details['file'], $uploads['path'] . '/' . $obj->filename );
-							register_shutdown_function( 'unlink', $uploads['path'] . '/' . $obj->filename );
-						}
-						$details['file'] = str_replace( basename( $post_url ), basename( $obj->filename ), $details['file'] );
-					}
+		if ( is_wp_error( $upload_result ) ) {
+			$details['error'] = $upload_result->get_error_message();
+		} else {
+			$result_filename = $upload_result;
+			if ( isset( $result_filename ) && basename( $result_filename ) != basename( $post_url ) ) {
```

Comment:
Is this block still necessary?

Reply:
I think you are correct. At first glance, it looked like it was doing more with $post_url but I think most of that could be refactored out. I was a bit concerned with breaking existing functionality since the existing logic above and below the curl call was a bit tricky to get a complete understanding of. I think some tests around that could help with the understanding and refactoring.